src/transaction/log_replication_atomic.cpp

```diff
@@ -216,6 +217,28 @@ namespace cublog
 	      }
 	    break;
 	  }
+	  case LOG_SCHEMA_MODIFICATION_LOCK:
+	  {
+	    char *classname = NULL;
+
+	    m_redo_context.m_reader.advance_when_does_not_fit (sizeof (LOG_REC_SCHEMA_MODIFICATION_LOCK));
+	    const LOG_REC_SCHEMA_MODIFICATION_LOCK log_rec =
+		    m_redo_context.m_reader.reinterpret_copy_and_add_align<LOG_REC_SCHEMA_MODIFICATION_LOCK> ();
+
+	    /* TODO:
+	     * All these debug logging part will be removed.
+	     * Lock will be acquired for the class that log_rec.classoid indicates
+	     */
+	    int error = heap_get_class_name (&thread_entry, &log_rec.classoid, &classname);
```

Comment:
Can we guarantee that we can get the classname here? I'm asking because I remember that I encountered a tricky cases back when I treated the TRUNCATE improvement. When creating a table, it seems that sch_m_lock is acquired but the class record is not yet determined and it's determined later. 

In my quick analysis, when creating a table:

sch_m_lock is acquired at:
do_create_entity()->dbt_create_class()->dbt_reserve_name()->locator_reserve_class_name()->xlocator_reserve_class_names()-xlocator_reserve_class_name()->lock_object()

the classed seems to be assigned at:
do_create_entity()->dbt_finish_class()->sm_finish_class()->update_class()

Reply:
In this case, we probably even can't believe the oid. We don't leave a log record for every sch_m_lock?

Reply:
You're absolutely right. During the CREATE TABLE operation, we only have information about where the class record will be inserted (OID), and we don't know the name at that point. The class name information will only be available at the commit point. In this case, only the OID information will be logged, 
and whether this OID corresponds to the actual class OID or not can only be determined by examining the debuggig message from other DDL statements on the same table or by leaving one more debugging message at the commit point.

In the xlocator_reserve_class_names() function, it attempt to obtain SCH_M_LOCK using an invalid OID (-2, -2, -2). Therefore, in this case, we do not log the lock